salt/utils/parsers.py

```diff
@@ -654,7 +661,9 @@ def __setup_logfile_logger_config(self, *args):  # pylint: disable=unused-argume
             self.config.pop(self._logfile_loglevel_config_setting_name_)
```

Comment:
I think the above `if` statement may cause the tests failure. It's really a duplicate of the another `if` from `process_log_level_logfile` function, which already does popping the value form config if unset.